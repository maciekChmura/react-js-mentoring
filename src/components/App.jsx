import React, { Component } from 'react';
import styled from 'styled-components';
import PageName from './Header/PageName';
import GlobalStyle from './GlobalStyle';
import netflixBack from '../img/netflix_back.jpg';
import FormTitle from './Header/FormTitle';
import SearchForm from './Header/SearchForm';
import Results from './Body/Results';
import ResultsOptions from './Helper/ResultsOptions';
import MovieDetails from './Detail/MovieDetails';

const MainCSSGrid = styled.div`
  display: grid;
  grid-template-columns: 1fr 900px 1fr;
  grid-template-rows: 230px 50px auto 50px 500px;
`;

const HeaderCSSGrid = styled.div`
  grid-area: 2 / 3 / 1 / 2;
  display: grid;
  grid-template-columns: 50px 800px 50px;
  grid-template-rows: 80px 50px 100px;
  background-image: url(${netflixBack});
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
`;

const Footer = styled.div`
  grid-area: 4 / 2 / 5 / 3;
  background-color: lightgrey;
`;

const Detail = styled.div`
  grid-area: 5 / 2 / 6 / 3;
  background-color: black;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-image: url(${netflixBack});
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
`;

class App extends Component {
  state = {
    data: [],
    sortingType: 'release date',
    searchOption: 'title',
    movie: ''
  };

  performSearch = searchString => {
    const { sortingType, searchOption } = this.state;
    fetch(
      `http://react-cdp-api.herokuapp.com/movies?sortBy=${sortingType.replace(
        ' ',
        '_'
      )}&sortOrder=desc&search=${searchString}&searchBy=${searchOption}&limit=12`
    )
      .then(response => response.json())
      .then(data => this.setState({ data: data.data }));
  };

  componentDidMount = () => {
    fetch('http://react-cdp-api.herokuapp.com/movies?limit=12')
      .then(response => response.json())
      .then(data => this.setState({ data: data.data }));

    // temporary fetch, just to present movie details
    fetch('http://react-cdp-api.herokuapp.com/movies/12')
      .then(response => response.json())
      .then(data => this.setState({ movie: data }));
  };

  changeSorting = data => {
    console.log(data);
    this.setState({ sortingType: data });
  };

  changeSearch = data => {
    event.preventDefault(); // TODO probably bad solution
    this.setState({ searchOption: data });
  };

  render() {
    const { data, sortingType, searchOption, movie } = this.state;
    return (
      <MainCSSGrid>
        <GlobalStyle />
        <HeaderCSSGrid>
          <PageName />
          <FormTitle />
          <SearchForm
            handleFormSubmit={this.performSearch}
            searchOption={searchOption}
            changeSearch={this.changeSearch}
          />
        </HeaderCSSGrid>
        <ResultsOptions
          dataSize={data.length}
          changeSorting={this.changeSorting}
          sortingType={sortingType}
        />
        {data ? <Results results={data} /> : <p>loader</p>}
        <Footer />
        <Detail>
          {movie ? <MovieDetails details={movie} /> : <p>loader</p>}
        </Detail>
      </MainCSSGrid>
    );
  }
}

export default App;
