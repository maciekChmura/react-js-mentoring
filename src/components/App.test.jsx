import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('App', () => {
  it('should render correctly', () => {
    const component = shallow(<App />);

    expect(component).toMatchSnapshot();
  });

  it('should render Search as default', () => {
    const component = shallow(<App />);
    expect(component.find('SearchPage')).toHaveLength(1);
    expect(component.find('DetailPage')).toHaveLength(0);
  });

  it('should render Detail when state.pageType is detail', () => {
    const component = shallow(<App />);
    expect(component.find('SearchPage')).toHaveLength(1);
    expect(component.find('DetailPage')).toHaveLength(0);
    component.setState({ pageType: 'detail' });
    expect(component.find('SearchPage')).toHaveLength(0);
    expect(component.find('DetailPage')).toHaveLength(1);
  });

  it('should render ChangePageButton', () => {
    const component = shallow(<App />);
    expect(component.find('ChangePageButton')).toHaveLength(1);
  });

  it('changePage changes state when called', () => {
    const component = shallow(<App />);

    component.find('ChangePageButton').prop('changePage')();
    expect(component.state('pageType')).toEqual('detail');
    component.find('ChangePageButton').prop('changePage')();
    expect(component.state('pageType')).toEqual('search');
  });
});
