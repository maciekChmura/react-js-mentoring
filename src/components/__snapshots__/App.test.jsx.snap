// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
Array [
  <div
    className="SearchPageStyles__SearchCSSGrid-sc-1ueu3tk-0 bsBxtI"
  >
    <div
      className="SearchPageStyles__HeaderCSSGrid-sc-1ueu3tk-1 dKHhZy"
    >
      <h1
        className="PageNameStyles__H1-sc-1qqo73k-0 fMwOSL"
      >
        Netflix Roulette
      </h1>
      <h2
        className="FormTitleStyles__H2-sc-10gmsuy-0 koCUjc"
      >
        FIND YOUR MOVIE
      </h2>
      <div
        className="SearchFormStyles__Wrapper-emg76u-0 gbhWJX"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            className="SearchFormStyles__InputWrapper-emg76u-1 iBOVkC"
            id="location"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="type something"
            value=""
          />
          <div
            className="SearchFormStyles__ButtonsWrapper-emg76u-2 ideRDc"
          >
            <p
              className="SearchFormStyles__Display-emg76u-3 iyaqPI"
            >
              Search by: 
            </p>
            <button
              className="OptionButtonStyles__Button-sc-1yyses4-0 eKWgln"
              onClick={[Function]}
            >
              title
            </button>
            <button
              className="OptionButtonStyles__Button-sc-1yyses4-0 hkNaKA"
              onClick={[Function]}
            >
              genres
            </button>
            <button
              className="SearchFormStyles__SearchButton-emg76u-4 erPeEk"
              onClick={[Function]}
              type="button"
            >
              Search
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      className="ResultsOptionsStyles__ResultsOptionsWrapper-qi5qz4-0 bcyhbG"
    >
      <p
        className="ResultsOptionsStyles__Display-qi5qz4-2 gCbovo"
      >
        0
         movies found
      </p>
      <div
        className="ResultsOptionsStyles__ButtonsWrapper-qi5qz4-1 cVcvWT"
      >
        <p
          className="ResultsOptionsStyles__Display-qi5qz4-2 gCbovo"
        >
          Sort by:
        </p>
        <button
          className="OptionButtonStyles__Button-sc-1yyses4-0 eKWgln"
          onClick={[Function]}
        >
          release date
        </button>
        <button
          className="OptionButtonStyles__Button-sc-1yyses4-0 hkNaKA"
          onClick={[Function]}
        >
          rating
        </button>
      </div>
    </div>
    <p>
      loading
    </p>
  </div>,
  <div
    className="AppStyles__ChangePageWrapper-sc-166dlhi-4 ilRoMU"
  >
    <button
      className="ChangePageButtonStyles__Button-ws550g-0 famkUL"
      onClick={[Function]}
    >
      change page
    </button>
  </div>,
]
`;
